### ISSUE TITLE
feature wish: non inversion-of-control interface for processing query rows

### ISSUE THREAD

**Issue by gdsfh** [ygrek/sqlgg#15]
For example, Stream.t of rows, where row = object or record or tuple of column values.

**Comment by ygrek** [comment]
I planned to do it but the problem is that there is no universally adopted sequence type, do you think `Stream` is a good candidate or better functorize it?

**Comment by gdsfh** [comment]
`Stream` is not "adopted": poor stdlib support, people don't like it.  But now I have an idea: what if sqlgg will generate `(myquery db ~args) : 'a -> row option`?  `'a` will be unified with `int` when this function will be passed to `Stream.from`, also the type (and semantics) are generic enough to be used with any sequences/generators/producers/consumers.

**Comment by rgrinberg** [comment]
I agree that the "gen" like type proposed @gdsfh is best. There's already a lib for handling this kind of stream: https://github.com/c-cube/sequence/blob/master/sequence.mli#L447
